### DISCUSSION TITLE
Lookup forge_remote_id by repo path

### DISCUSSION THREAD

**Discussion by tyler71** [woodpecker-ci/woodpecker#4231] [Ask a Question] [answered]
Hello, here is what I am trying to do:
Create a repo at GitHub, send an API request to activate that repo at Woodpecker-CI

My expected behavior:

1. Lookup `forge_remote_id` by full name lookup
```text
GET /api/repos/lookup/user/reponame
```
2. Activate the repository
```text
POST /repos?forge_remote_id=123
```

However, `/api/repos/lookup/user/reponame` returns a 404, _unless_ it's already enabled.

---

Looking at the network requests in the UI, when I try to add a repo, it pulls all repo's for my user.
Is this the only way to do it?
```text
GET /api/user/repos?all=true
```

**Comment by anbraten** [comment]
`GET /api/repos/lookup/user/reponame` is only working for activated repos.

> Is this the only way to do it?

I think so. I might be possible to get the `forge_remote_id` directly from github however as it is the id github uses internally

**Comment by tyler71** [comment, marked as answer]
Thanks for the tip, I will look it up independently for now.
```javascript
  const res = await fetch(`https://api.github.com/repos/${user}/${reponame}`, {
    headers: { Authorization: `Bearer ${token}` }
  })
 const resJson = await res.json()
 console.log({forgeId: resJson.id})
```
